File: app/src/main/java/com/example/proyectocurrencyexchangepersonal/database/AppDatabase.java
package com.example.proyectocurrencyexchangepersonal.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.proyectocurrencyexchangepersonal.daos.UsuarioDao;
import com.example.proyectocurrencyexchangepersonal.daos.MonedaDao;
import com.example.proyectocurrencyexchangepersonal.daos.TransaccionMonedaDao;
import com.example.proyectocurrencyexchangepersonal.models.Usuario;
import com.example.proyectocurrencyexchangepersonal.models.Moneda;
import com.example.proyectocurrencyexchangepersonal.models.TransaccionMoneda;

@Database(entities = {Usuario.class, Moneda.class, TransaccionMoneda.class}, version = 2, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase INSTANCE;

    public abstract UsuarioDao usuarioDao();
    public abstract MonedaDao monedaDao();
    public abstract TransaccionMonedaDao transaccionMonedaDao();

    public static synchronized AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "currency_exchange_db")
                    .fallbackToDestructiveMigration() // Opcional, destruye y crea si cambia versión
                    .build();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/proyectoexamencrudbd/data/AppDatabase.kt
package com.example.proyectoexamencrudbd.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.proyectoexamencrudbd.data.dao.CriaturaMagicaDao
import com.example.proyectoexamencrudbd.data.dao.FichaCuidadoDao
import com.example.proyectoexamencrudbd.data.dao.HabitatDao
import com.example.proyectoexamencrudbd.data.entity.CriaturaMagica
import com.example.proyectoexamencrudbd.data.entity.FichaCuidado
import com.example.proyectoexamencrudbd.data.entity.Habitat

@Database(
    entities = [
        Habitat::class,
        CriaturaMagica::class,
        FichaCuidado::class
    ],
    version = 1,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun habitatDao(): HabitatDao
    abstract fun criaturaMagicaDao(): CriaturaMagicaDao
    abstract fun fichaCuidadoDao(): FichaCuidadoDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "criaturas_magicas_db"
                )
                .fallbackToDestructiveMigration()
                .build()
                INSTANCE = instance
                instance
            }
        }
    }
} 